akka-remote/src/main/resources/reference.conf

```diff
@@ -98,6 +98,11 @@ akka {
 
     # Acknowledgment timeout of management commands sent to the transport stack.
     command-ack-timeout = 30 s
+    
+    # The timeout for outbound associations to perform the handshake.
+    # If the transport is akka.remote.netty.tcp or akka.remote.netty.ssl
+    # the configured connection-timeout for the transport will be used instead.
```

Comment:
I remember a customer issue in which SSL was taking so long to init that connections timed out. So from that perspective, if read this PR correctly it would be easier to tune for their case as well.

The error message would also have been nicer, so I'm for this PR, though maybe I'm missing some more fuzzy interplay with other remoting things